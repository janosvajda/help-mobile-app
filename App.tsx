import { StatusBar } from 'expo-status-bar';
import {
  StyleSheet, Text, View, Image, TextInput,
  TouchableOpacity
} from 'react-native';
import React, { useState } from 'react';

export default function App() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const buttonClickedHandler = () => {
    console.log('You have been clicked a button!');
    // do something
  };
  return (
    <View style={styles.container}>
      <Image
        source={{ uri: 'https://github.com/janosvajda/help-mobile-app/blob/main/assets/icon.png' }}
        style={{ width: 40, height: 40 }}
      />
      <Text style={styles.title}>!HÆŽlp!</Text>
      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="Email"
          placeholderTextColor="#003f5c"
          onChangeText={(email) => setEmail(email)}
        />
      </View>

      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="Password"
          placeholderTextColor="#003f5c"
          secureTextEntry={true}
          onChangeText={(password) => setPassword(password)}
        />
      </View>
      <View style={{ flexDirection: "row" }}>
        <View style={styles.buttonStyle}>
          <TouchableOpacity
            onPress={buttonClickedHandler}
            style={styles.roundButton}>
            <Text>Login</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.buttonStyle}>
          <TouchableOpacity
            onPress={buttonClickedHandler}
            style={styles.roundButton}>
            <Text>SignUp</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#73A9AD',
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    marginBottom: 40,
    height: 100,
    width: 100
  },
  title: {
    color: 'white',
    fontSize: 36,
    fontWeight: 'bold',
    marginBottom: '30%',
    fontFamily: 'MMA Champ',
  },
  loginText: {
    color: 'white',
    fontSize: 12,
    fontWeight: 'bold',
    marginBottom: '30%',
    fontFamily: 'MMA Champ',
  },
  inputView: {
    backgroundColor: "#F8F9D7",
    borderRadius: 30,
    width: "70%",
    height: 45,
    marginBottom: 20,

    alignItems: "center",
  },
  TextInput: {
    height: 50,
    flex: 1,
    padding: 10,
    marginLeft: 20,
  },
  roundButton: {
    width: 160,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    borderRadius: 12,
    backgroundColor: '#C4D7E0',
  },
  buttonStyle: {
    margin: 4
  }
});
