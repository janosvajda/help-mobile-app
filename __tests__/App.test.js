import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

describe('<App />', () => {
  it('has 5 childs', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(5);
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});