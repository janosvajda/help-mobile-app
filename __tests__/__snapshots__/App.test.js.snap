// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 5 childs 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#73A9AD",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://github.com/janosvajda/help-mobile-app/blob/main/assets/icon.png",
      }
    }
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "MMA Champ",
        "fontSize": 36,
        "fontWeight": "bold",
        "marginBottom": "30%",
      }
    }
  >
    !Help
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F8F9D7",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#003f5c"
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F8F9D7",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#003f5c"
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "margin": 4,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C4D7E0",
            "borderRadius": 12,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
            "width": 160,
          }
        }
      >
        <Text>
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 4,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C4D7E0",
            "borderRadius": 12,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
            "width": 160,
          }
        }
      >
        <Text>
          SignUp
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<App /> match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#73A9AD",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://github.com/janosvajda/help-mobile-app/blob/main/assets/icon.png",
      }
    }
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "MMA Champ",
        "fontSize": 36,
        "fontWeight": "bold",
        "marginBottom": "30%",
      }
    }
  >
    !Help
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F8F9D7",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#003f5c"
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F8F9D7",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#003f5c"
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "margin": 4,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C4D7E0",
            "borderRadius": 12,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
            "width": 160,
          }
        }
      >
        <Text>
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 4,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C4D7E0",
            "borderRadius": 12,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
            "width": 160,
          }
        }
      >
        <Text>
          SignUp
        </Text>
      </View>
    </View>
  </View>
</View>
`;
